src/rgw/rgw_kms.cc

```diff
@@ -23,6 +24,15 @@ static map<string,string> get_str_map(const string &str) {
   return m;
 }
 
+static void concat_url(std::string &url, std::string path) {
+  if (url.back() == '/' && path.front() == '/') {
```

Comment:
this assumes `url` and `path` are both non-empty, so we'll either need checks here or at the call sites

Reply:
I think the only problem here is if both url and path are empty (the result would be a single slash "/"). In practice this can't happen here since in both cases where it's used (Barbican and Vault functions) we're already checking if the base URL is empty. In any case, I'm adding add a check to the function to prevent any issues in the future, and also some unit tests.